core/src/lightclient/mod.rs

```diff
@@ -82,4 +122,61 @@ impl StarknetClient {
         let block = starknet_block_to_eth_block(starknet_block);
         Ok(block)
     }
+
+    /// Get the number of transactions in a block given a block id.
+    /// The number of transactions in a block.
+    ///
+    /// # Arguments
+    ///
+    ///
+    ///
+    /// # Returns
+    ///
+    ///  * `block_number(u64)` - The block number.
+    ///
+    /// `Ok(Bytes)` if the operation was successful.
+    /// `Err(LightClientError)` if the operation failed.
+    async fn get_code(
+        &self,
+        ethereum_address: Address,
+        starknet_block_id: StarknetBlockId,
+    ) -> Result<Bytes, LightClientError> {
+        // Convert the Ethereum address to a hex-encoded string
+        let address_hex = hex::encode(ethereum_address);
+        // Convert the hex-encoded string to a FieldElement
+        let ethereum_address_field_element = FieldElement::from_hex_be(&address_hex).unwrap();
+
+        // Prepare the calldata for the get_starknet_contract_address function call
+        let tx_calldata_vec = vec![ethereum_address_field_element];
+        let request = FunctionCall {
+            contract_address: self.kakarot_account_registry,
+            entry_point_selector: selector!("get_starknet_contract_address"),
+            calldata: tx_calldata_vec,
+        };
+        // Make the function call to get the Starknet contract address
+        let starknet_contract_address = self.client.call(request, &starknet_block_id).await?;
+        // Concatenate the result of the function call
+        let concatenated_result = starknet_contract_address
+            .into_iter()
+            .fold(FieldElement::ZERO, |acc, x| acc + x);
+
+        // Prepare the calldata for the bytecode function call
+        let tx_calldata_vec2 = vec![];
+        let request = FunctionCall {
+            contract_address: concatenated_result,
+            entry_point_selector: selector!("bytecode"),
```

Comment:
selector inside selectors.rs:)!

later one we could even create script that takes an ABI and generates starknet-rs selectors:)! 
but later!